query_builder_test.go

```diff
@@ -3,8 +3,158 @@ package qb
 import (
 	"reflect"
 	"testing"
+	"time"
 )
 
+type testInterface interface {
+	qbTable()
+}
+
+type testTable struct {
+	ID    string `dbtable:"users" db:"id" table:"foo" col:"foo_id"`
+	Name  string `db:"name" col:"foo_name"`
+	Email string `db:"email" col:"foo_email"`
+}
+
+func (t *testTable) qbTable() {}
+
+type testTableNoName struct {
+	ID    string `db:"id"`
+	Name  string `db:"name"`
+	Email string `db:"email"`
+}
+
+type testModel struct {
+	ID string `db:"id"`
+	TestModelWithTime
+}
+
+type TestModelWithTime struct {
+	CreatedAt time.Time `db:"created_at"`
+	DeletedAt time.Time `db:"deleted_at"`
+}
+
+type testModelType struct {
+	testModel `dbtable:"model"`
+	Name      string `db:"name"`
+	Email     string `db:"email"`
+}
+
+type testModelTypePtr struct {
+	*string
+	*testModel `dbtable:"model"`
+	Name       string `db:"name"`
+	Email      string `db:"email"`
+}
+
+func TestNew(t *testing.T) {
+	testTableInterface := func() testInterface {
+		return &testTable{}
+	}
+
+	s := "string"
+
+	type args struct {
+		i    interface{}
+		opts []Option
+	}
+	tests := []struct {
+		name    string
```

Comment:
This is personal preference but perhaps a map `name -> struct` would be better than inlining the name like this. Additionally, not sure why you need the `args` struct. Nesting `args` just makes things more verbose.

Reply:
Name is the test name, so if it fails, it shows you the test name, `FAIL: TestNew/ok`.

The tool `gotests` adds `args` and/or `fields` and both are verbose, but they are easier to maintain if new arguments or fields are added.

Reply:
What I'm saying is that a map would work better and would also have the benefit of shuffling the order of subtests ([ref](https://github.com/golang/go/wiki/TableDrivenTests#using-a-map-to-store-test-cases)]